pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -106,6 +108,7 @@ type Config struct {
 		ApiKey     string `gcfg:"api-key"`
 		TenantId   string `gcfg:"tenant-id"`
 		TenantName string `gcfg:"tenant-name"`
+		TrustId    string `gcfg:"trust-id"`
```

Comment:
Actually using a token here would defeat the main use case of trusts : if we just want to delegate rights for a small time window passing a token would be enough. However in the OpenStack driver use case Kubernetes is a long living service and its lifetime is way higher than the token lifetime (usually a day). In this case the trust feature allows to delegate user rights to another user, that would have been created specifically for the Kubernetes cluster. Hence we do not leak the user credentials, only the creds of the user specifically created, which we don't care. And we can specify the lifetime we want for the trust when creating it (can be unlimited).